superbench/benchmarks/micro_benchmarks/micro_base.py

```diff
@@ -40,17 +41,46 @@ def _preprocess(self):
 
     @abstractmethod
     def _benchmark(self):
-        """Implementation for benchmarking."""
+        """Implementation for benchmarking.
+
+        Return:
+            True if run benchmark successfully.
+        """
         pass
 
-    def _process_micro_result(self, output):
+    def _process_numeric_result(self, metric, result):
+        """Function to save the numerical results.
+
+        Args:
+            metric (str): metric name which is the key.
+            result (List[numbers.Number]): numerical result.
+
+        Return:
+            True if result list is not empty.
+        """
+        if len(result) == 0:
+            logger.error(
+                'Numerical result of benchmark is empty - round: {}, name: {}.'.format(
+                    self._curr_run_index, self._name
+                )
+            )
+            return False
+
+        self._result.add_raw_data(metric, result)
+        self._result.add_result(metric, sum(result) / len(result))
+        return True
+
+    def _process_raw_result(self, raw_output):
         """Function to process raw results and save the summarized results.
 
         Args:
-            output (str): raw output string of the micro-benchmark.
+            raw_output (str): raw output string of the micro-benchmark.
+
+        Return:
+            True if the raw output string is valid and result can be extracted.
```

Comment:
will this function save results? how does it know where to save?

Reply:
This function also need to call the following function to save result:
        self._result.add_raw_data()
        self._result.add_result()
       
Benchmark owner need to know this. I will add more comments.